src/main/java/hudson/plugins/analysis/core/AbstractResultAction.java

```diff
@@ -51,7 +58,7 @@
      * @param result
      *            the result of the action
      */
-    public AbstractResultAction(final AbstractBuild<?, ?> owner, final AbstractHealthDescriptor healthDescriptor, final T result) {
+    public AbstractResultAction(final Run<?, ?> owner, final AbstractHealthDescriptor healthDescriptor, final T result) {
```

Comment:
For compatibility you may need to keep the previous constructor around, deprecated.